crates/relayer/src/chain/cosmos.rs

```diff
@@ -577,22 +578,25 @@ impl CosmosSdkChain {
         let mut begin_block_events = vec![];
         let mut end_block_events = vec![];
 
-        for seq in request.sequences.iter() {
+        for seq in request.sequences.iter().copied() {
             let response = self
                 .block_on(self.rpc_client.block_search(
-                    packet_query(request, *seq),
+                    packet_query(request, seq),
+                    // We only need the first page
                     1,
-                    1, // there should only be a single match for this query
-                    Order::Ascending,
+                    // There should only be a single match for this query, but due to
+                    // the fact that the indexer treat the query as a disjunction over
+                    // all events in a block rather than a conjunction over a single event,
+                    // we may end up with partial matches and therefore have to account for
+                    // that by fetching multiple results and filter it down after the fact.
+                    10,
```

Comment:
I think in the worst case we get N blocks where N is the number of channels. Maybe clarify this.
Also should we have a `const` for this?